src/index.ts

```diff
@@ -48,22 +48,28 @@ const server = Bun.serve<ClientData>({
          if (roomManager.joinRoom(webSocket.data.roomId, webSocket.data.uuid)) {
             logger.info(`${webSocket.data.uuid} connected to room ${webSocket.data.roomId}`)
 
-            webSocket.publish(webSocket.data.roomId, new JoinPacket(webSocket.data.uuid).toString())
+            server.publish(webSocket.data.roomId, new JoinPacket(webSocket.data.uuid).toString())
             webSocket.subscribe(webSocket.data.roomId)
             webSocket.send(new WelcomePacket(webSocket.data.uuid, roomManager.getRoom(webSocket.data.roomId)).toString())
          } else {
             webSocket.close(1011, new ServerPacket("LIMIT", webSocket.data.roomId).toString())
          }
       },
-      message: function (webSocket: ServerWebSocket<ClientData>, message: string | Buffer): void | Promise<void> {
-         webSocket.publish(webSocket.data.roomId, message)
+      message: function (webSocket: ServerWebSocket<ClientData>, message: string): void | Promise<void> {
+         let packet = JSON.parse(message) as Packet;
+         if(packet !== undefined && packet.getType().startsWith("CLIENT_")) {
```

Comment:
1. packet cannot be undefined here.
When there is no valid json object in message JSON.parse throws an error.

2. it is not possible for packet to have a function "getType"
JSON.parse always just returns a plain object, the parsed JSON. And since you cannot define a function in JSON you cannot have a function in a parsed JSON object.
Therefore this code is broken and will always throw an error